### PR TITLE
ARROW-7570: [Java] Fix high severity issues

### PR DESCRIPTION
**Pull request by Fokko**
Fixes two high severity issues reported by LGTM: https://lgtm.com/projects/g/apache/arrow/?mode=list&lang=java&severity=error

### CODE REVIEW

File: java/memory/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java [hunk 1]
```diff
@@ -268,4 +267,19 @@ public int hashCode() {
...
 
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+
+    if (obj instanceof UnsafeDirectLittleEndian) {
+      UnsafeDirectLittleEndian that = (UnsafeDirectLittleEndian) obj;
+      return this.wrapped.equals(that.wrapped) &&
```
Comment (jhorstmann) [inline java/memory/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java]: This does not seem to be consistent with `hashCode`, which is based on object identity. To be consistent, `equals` should just return `this == obj`.

Comment (Fokko) [reply java/memory/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java]: Good point, thanks

**Review by siddharthteotia — APPROVED** [review]

**Comment by siddharthteotia** [comment]
Thanks @Fokko . It would be good to mention in the PR description itself the nature of issues reported by static analysis.

**Comment by emkornfield** [comment]
I'm going to merge.  @Fokko do you know of a way of adding these checks as part of CI?

**Comment by Fokko** [comment]
Yes, there is a LGTM plugin available for Github. Apache Druid is using this.

**Pull request closed without merging**